### PR TITLE
Added psisloo & loo_compare

### PR DESCRIPTION
**Pull request by jburos**
Added psisloo & loo_compare functions, as thin wrappers around Aki Vehtari's [PSIS repo ](https://github.com/avehtari/PSIS/blob/master/py/psis.py). References issue #5. 

Currently imports this code using a git submodule. Would be better to modify Aki's repo to enable direct import from git via `pip`, but git-submodule should be OK for now. 

For now, since we are using git submodules, best practice is as follows : 
- Use `git clone --recursive` when checking out this repo
- Or, to run `git submodule update --init --recursive` after checkout / update

### CODE REVIEW

File: stanity/psisloo.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
+
+import psis
+import pandas
+import math
+import numpy
+import seaborn
+import matplotlib as pyplot
```
Comment (timodonnell) [inline stanity/psisloo.py]: looks like this is unused anyway so should be dropped, but fwiw pyplot is a module within matplotlib, so you'd want `from matplotlib import pyplot`

File: stanity/psisloo.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        self.summary['greater than 1'] = self.summary.pareto_k > 1
+
+    def print_summary(self):
+        return self.summary.apply(numpy.mean)[2:]
+
+    def plot(self):
+        seaborn.pointplot(y = self.pointwise.pareto_k, x = self.pointwise.index, join = False)
+        #pyplot.axhline(0.5)
+
+def psisloo(log_lik, *args, **kwargs):
```
Comment (timodonnell) [inline stanity/psisloo.py]: args and kwargs are unused

File: stanity/psisloo.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    For the method to be valid, the two models should have been fit on the same input data. 
+
+    Parameters
+    -------------------
+    psisloo1 : Psisloo object for model1
+    psisloo2 : Psisloo object for model2
+
+    Returns
+    -------------------
+    Dict summarizing difference between two models, where a positive value indicates
```
Comment (timodonnell) [inline stanity/psisloo.py]: might be helpful to mention what the keys are in the dict

File: setup.py [hunk 4]
```diff
@@ -34,5 +34,7 @@
         "future>=0.14.3",
         "pandas",
         "seaborn",
-    ]
+        "matplotlib",
+	"numpy",
```
Comment (timodonnell) [inline setup.py]: nit: we've standardized on spaces instead of tabs

File: stanity/psisloo.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
+
+import psis
+import pandas
+import math
+import numpy
+import seaborn
+import matplotlib as pyplot
+
+class Psisloo(object):
+    def __init__(self, log_lik):
```
Comment (timodonnell) [inline stanity/psisloo.py]: A short description of what each of these attributes are in the docstring for the class would be helpful

Also I'd be biased toward non-abbreviated names (e.g. `log_likelihood`) unless you think the abbreviation is so standard that spelling it out would be more confusing (I haven't used the R libraries so I'm not sure if that is the case for these names)

**Comment by timodonnell** [comment]
LGTM, thanks for adding this @jburos ! I added some very minor comments.

I agree that the submodule is ok for now but would be good to either depend on psis.py as a pip package, or copy it into this repo (whichever @avehtari would prefer is fine with me, and by the way thanks for implementing that, @avehtari !).

For some minor code style stuff, if you are using SublimeText you may want to try the `SublimeLinter-pyflakes` and `SublimeLinter-pep8` packages. I've found that setup to be pretty helpful at keeping style consistent.

**Comment by avehtari** [comment]
Great! Two days ago @jgabry wrote in an email that some Devin at Columbia could make a proper PSIS-LOO Python package based on PSIS repo soon. We'll let you know when that's ready.

File: stanity/psisloo.py (hunk 1 above)
Comment (jburos) [inline stanity/psisloo.py]: Thanks - dropped this

File: stanity/psisloo.py (hunk 2 above)
Comment (jburos) [inline stanity/psisloo.py]: Thanks. Removed these.

File: stanity/psisloo.py (hunk 3 above)
Comment (jburos) [inline stanity/psisloo.py]: Added this detail

File: setup.py (hunk 4 above)
Comment (jburos) [inline setup.py]: GTK. Didn't notice these got in there. fixed

**Comment by jburos** [comment]
Thanks @timodonnell for the detailed suggestions. Think I implemented them all.

As a last check, I tested a clean install of this branch via pip & got the following error : 

```
$ pip install git+git://github.com/hammerlab/stanity.git@dev
Collecting git+git://github.com/hammerlab/stanity.git@dev
  Cloning git://github.com/hammerlab/stanity.git (to dev) to /tmp/pip-1I7Pqt-build
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Clone of 'git@github.com:avehtari/PSIS.git' into submodule path 'stanity/PSIS' failed
Command "git submodule update --init --recursive -q" failed with error code 1 in /tmp/pip-1I7Pqt-build
```

I will troubleshoot. Once this works reliably I believe we'll be good to go. LMK if you have thoughts in the meantime.

**Pull request merged**